### PR TITLE
feat: enable user to share a link to the map that will open a drawer at a specific area id

### LINKED ISSUE #1209: Store selected area id in url for /maps endpoint
**What would you like to be able to do? Please describe.**
This would allow the user to share a map link that opens in a specification and opened popup.

**How important is this to you** (Please pick one)
- Not important
- Nice-to-have
- Important
- Critical

### PR DESCRIPTION
**Pull request by clintonlunn**
---
name: Pull request
about: Create a pull request
title: ''
labels: ''
assignees: ''

---

## What type of PR is this?(check all applicable)
- [ ] refactor
- [x] feature
- [ ] bug fix
- [ ] documentation
- [ ] optimization
- [ ] other

## Description

Previously we did not have a way to share a specific area id in the map view. This PR adds whichever area is currently active in the drawer into the the url. This enables the user to either click share on the url, or click a share button in the drawer which accomplishes the same thing. 

### Related Issues

Issue #1209 

### Screenshots, recordings

image

### Notes

In the future, I'd like to extend this to also the climb id, where the page will open with the relevant area drawer open, and the climb highlighted.

### CODE REVIEW

**Comment by clintonlunn** [comment]
@vnugent i think this PR is about complete. I made sure I am up to date with develop. This required some refactoring of existing camera params, which i've combined into a single useUrlParams in order to deal with the url in a single place. In addition, I've removed the debouncing that was happening for the camera movement, and instead change the url `onMoveEnd`, which I think gives us more expected behavior.

File: src/components/ui/ShareButton.tsx
Comment (vnugent) [inline src/components/ui/ShareButton.tsx:1]: Can we reuse the Share Link button on the climb/area page?

https://github.com/OpenBeta/open-tacos/blob/develop/src/app/(default)/components/AreaAndClimbPageActions.tsx#L39

File: src/app/(maps)/components/FullScreenMap.tsx [hunk 1]
```diff
@@ -1,45 +1,118 @@
...
   useEffect(() => {
-    const initialStateFromUrl = cameraParams.fromUrl()
+    if (isInitialized) return
 
-    if (initialStateFromUrl != null) {
-      setInitialCenter([initialStateFromUrl.center.lng, initialStateFromUrl.center.lat])
-      setInitialZoom(initialStateFromUrl.zoom)
+    const { camera, areaId: urlAreaId } = urlParams.fromUrl()
+
+    if (urlAreaId !== null) {
```
Comment (vnugent) [inline src/app/(maps)/components/FullScreenMap.tsx]: We typically use `UrlAreaId != null` (single equal sign) to test for nullness and undefined

File: src/js/hooks/useUrlParams.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+const useUrlParams = (): UseUrlParamsReturn => {
+  const pathname = usePathname()
+  const searchParams = useSearchParams()
+
+  const toUrl = ({ camera, areaId }: UrlProps): string => {
+    const params = new URLSearchParams()
+
+    if (areaId !== null && areaId !== undefined) {
```
Comment (vnugent) [inline src/js/hooks/useUrlParams.tsx]: more concise `if (areaId != null)`

**Review by vnugent — CHANGES_REQUESTED** [review]
I made some inline comments.  It's great that you refactor the hooks to its own file. Not high priority (if you can do it in follow up PR) is to clear `areaId` when the user clicks elsewhere on the map (currently we close the drawer, clearing the current selected area). I think this behavior is consistent with other popular maps.

File: src/components/maps/GlobalMap.tsx [hunk 3]
```diff
@@ -43,68 +45,74 @@ interface GlobalMapProps {
...
     organizations: false,
     heatmap: false,
     crags: true
   })
+  const router = useRouter()
+  const urlParams = useUrlParams()
 
   const setActiveFeatureVisual = (feature: ActiveFeature | null, fState: FeatureState): void => {
-    if (feature == null || mapInstance == null) return
+    if (feature === null || mapInstance === null) return
```
Comment (vnugent) [inline src/components/maps/GlobalMap.tsx]: Can you change triple `===` to double catch both null and undefined?  (there are several instances).  Other than that I'm good to merge.

**Review by vnugent — CHANGES_REQUESTED** [review]

**Pull request merged**